addons/project/models/project.py

```diff
@@ -1289,6 +1290,10 @@ def _search_is_private(self, operator, value):
         else:
             return [('project_id', '!=', False)]
 
+    @api.onchange('project_id')
+    def _onchange_kanban_state(self):
```

Comment:
```suggestion
    @api.onchange('project_id')
    def _onchange_project_id(self):
```
The onchange methods should be named after the field invoking them: https://www.odoo.com/documentation/master/developer/misc/other/guidelines.html?highlight=onchange#symbols-and-conventions